base/strings/basic.jl

```diff
@@ -610,9 +610,28 @@ julia> replace("abcdeγfgh", !isascii=>' ') # replace non-ASCII chars with space
 ```
 """
 isascii(c::Char) = bswap(reinterpret(UInt32, c)) < 0x80
-isascii(s::AbstractString) = all(isascii, s)
 isascii(c::AbstractChar) = UInt32(c) < 0x80
 
+function _isascii(bytes, first, last)
+    r = true
+    for n = first:last
+        @inbounds r &= bytes[n] < UInt8(0x80)
+    end
+    return r
+end
+
+function isascii(s::AbstractString)
+    chunk_size = 1024
+    bytes = codeunits(s)
+    l = ncodeunits(s)
+    l < chunk_size && return _isascii(bytes, 1, l)
+    for n = 1:chunk_size:l
+        _isascii(bytes, n, n + chunk_size - 1) || return false
```

Comment:
I think this leads to an out-of-bounds access to `bytes` in `_isascii` for the last chunk unless `l` is a multiple of `chunk_size`.

Reply:
Indeed - run with `julia --check-bounds=yes`:

```
julia> isascii("a"^1024 * "ab")
ERROR: BoundsError: attempt to access 1026-codeunit String at index [1027]
Stacktrace:
 [1] checkbounds
   @ ./strings/basic.jl:216 [inlined]
 [2] codeunit
   @ ./strings/string.jl:117 [inlined]
 [3] getindex
   @ ./strings/basic.jl:756 [inlined]
 [4] _isascii(bytes::Base.CodeUnits{UInt8, String}, first::Int64, last::Int64)
   @ Main ./REPL[1]:4
 [5] isascii(s::String)
   @ Main ./REPL[4]:7
 [6] top-level scope
   @ REPL[5]:1
```

Reply:
Changed ` _isascii(bytes, n, n + chunk_size - 1)` -> ` _isascii(bytes, n, min(l,n + chunk_size - 1))`